### ISSUE TITLE
SemgrepLspServer doesn't need to implement LspServer

### ISSUE THREAD

**Issue by alexander-doroshko** [semgrep/semgrep-intellij#33]
Hi from JetBrains! 

I noticed that the plugin has the class `SemgrepLspServer`, which implements `LspServer`.

As I understand, this is a trivial class that only helps its users not to write `lsp4jServer as SemgrepLanguageServer` because it overrides it like this:
```
    override val lsp4jServer: SemgrepLanguageServer
        get() = server.lsp4jServer as SemgrepLanguageServer
```

May I suggest that you refactor this class not to implement `LspServer`?

This would be more accurate from the architectural point of view, and, what's more important, your plugin will stay compatible, when we change the API of the `LspServer` interface by adding more properties or functions.

Instead of implementing it, the `SemgrepLspServer` class could expose the original `LspServer` instance as a public property.

**Comment by ajbt200128** [comment]
Yea that makes sense, thanks for the suggestion!

**Comment by alexander-doroshko** [comment]
It's clear now that the `LspServer` interface will be changed in 2024.1.
You can already make your plugin future-compatible by not implementing it.

**Comment by ajbt200128** [comment]
Closed by #36
